CMSgov/design-system
# [WNMGDS-2613] Updates drawer guidance

## Summary

Updates the `Drawer` and `HelpDrawer` guidance. See [jira ticket](https://jira.cms.gov/browse/WNMGDS-2613) for context.

### Changes Made
- Clarified definitions of `Drawer` and `HelpDrawer` in the overview section, moving accessibility-related content to the appropriate sections for better organization.
- Consolidated redundant bullets in the 'When to use' section and added a specific use case for `HelpDrawer` to clearly distinguish between the two components.
- Streamlined the guidance in the 'When to consider alternatives' section to remove overlap and make each scenario clearer.

### Follow-Up Actions
- Investigate the potential for a dedicated page on how to write effective links, given how often links are used across different components. This was brought up in the original [Jira ticket](https://jira.cms.gov/browse/WNMGDS-2613), but I thought it could be extracted into its own side quest.
- Add a section referencing the web component version of `Drawer` in Storybook. Since we have several React components with web component counterparts, we may want to consider creating a `WebComponentDocsLink`.

## How to test

Check it out [here](https://cmsgov.github.io/design-system/branch/tamara/WNMGDS-2613/update-drawer-guidance/components/drawer/?theme=core).

## Checklist
